# Fatals Police Shooting In The USA Analysis
### Sagi Polaczek
#### -14 September 2020- 


## Introduction
Police shooting is one of the most hot topics in the USA nowdays.  
.  
.  
.  


## Data Set
The dataset (fps_data.csv) was obtained from The Washington Post's repository.  
The Post's dataset contains records of every fatal shooting in the United States by a police officer in the line of duty since Jan. 1, 2015.  
The dataset describes 15 details about each killing, and is documenting only those shootings in which a police officer, in the line of duty, shoots and kills a civilian.  
.  
.  
.  


# Preliminaries:  
In this project we'll use the following python packages:

In [17]:
# Importing libraries we'll use
import numpy as np
import pandas as pd
import matplotlib as plt
import seaborn as sns

Lets import the data as a Pandas's DataFrame and take a first look:

In [32]:
fps_data = pd.read_csv('fps_data.csv', index_col = 'id')
print("\nThe dataset contains %d rows." %fps_data.shape[0])
print("\nThe dataset's columns:", list(fps_data.columns))
fps_data.head()


The dataset contains 5624 rows.

The data's columns: ['name', 'date', 'manner_of_death', 'armed', 'age', 'gender', 'race', 'city', 'state', 'signs_of_mental_illness', 'threat_level', 'flee', 'body_camera', 'longitude', 'latitude', 'is_geocoding_exact']


name        date   manner_of_death       armed   age gender  \
id                                                                              
3           Tim Elliot  2015-01-02              shot         gun  53.0      M   
4     Lewis Lee Lembke  2015-01-02              shot         gun  47.0      M   
5   John Paul Quintero  2015-01-03  shot and Tasered     unarmed  23.0      M   
8      Matthew Hoffman  2015-01-04              shot  toy weapon  32.0      M   
9    Michael Rodriguez  2015-01-04              shot    nail gun  39.0      M   

   race           city state  signs_of_mental_illness threat_level  \
id                                                                   
3     A        Shelton    WA                     True       attack   
4     W          Aloha    OR                    False       attack   
5     H        Wichita    KS                    False        other   
8     W  San Francisco    CA                     True       attack   
9     H          Evans    CO                    False       attack   

           flee  body_camera  longitude  latitude  is_geocoding_exact  
id                                                                     
3   Not fleeing        False   -123.122    47.247                True  
4   Not fleeing        False   -122.892    45.487                True  
5   Not fleeing        False    -97.281    37.695                True  
8   Not fleeing        False   -122.422    37.763                True  
9   Not fleeing        False   -104.692    40.384                True

Exploring missing and unique values:

In [27]:
print(fps_data.isnull().sum())

name                       209
date                         0
manner_of_death              0
armed                      212
age                        252
gender                       1
race                       591
city                         0
state                        0
signs_of_mental_illness      0
threat_level                 0
flee                       276
body_camera                  0
longitude                  272
latitude                   272
is_geocoding_exact           0
dtype: int64


Because the dataset has a total of 5624 rows, we are not concern by the missing values.   
Furthermore, in this project we focus on data exploration and visualization and not on building a prediction model, so we won't touch those missing values.

# Data Exploration
We'll start explored the data for each feature individually.   
Later on we'll procceed to multivariate visualization.